python/sglang/test/test_programs.py

```diff
@@ -517,3 +517,56 @@ def few_shot_hellaswag(s, question, choices):
     accuracy = np.mean(np.array(preds) == np.array(labels))
 
     return accuracy, latency
+
+
+def test_gen_min_new_tokens(model_path):
+    """
+    Validate sgl.gen(min_tokens) functionality.
+
+    The test asks a question where, without a min_tokens constraint, the generated answer is expected to be short.
+    By enforcing the min_tokens parameter, we ensure the generated answer has at least the specified number of tokens.
+    We verify that the number of tokens in the answer is >= the min_tokens threshold.
+    """
+    import sglang as sgl
+    from sglang.srt.hf_transformers_utils import get_tokenizer
+
+    MIN_TOKENS, MAX_TOKENS = 64, 128
+
+    @sgl.function
+    def convo_1(s):
+        s += sgl.user("What is the capital of the United States?")
+        s += sgl.assistant(
+            sgl.gen("answer", min_tokens=MIN_TOKENS, max_tokens=MAX_TOKENS)
+        )
+
+    @sgl.function
+    def convo_2(s):
+        s += sgl.user("What is the most beautiful integar?")
+        s += sgl.assistant(
+            sgl.gen_int("answer", min_tokens=MIN_TOKENS, max_tokens=MAX_TOKENS)
```

Comment:
`gen_int` uses `regex` under the hood. I think there is some confliction between `min_tokens` and `regex`. Do you need to use them together?

Reply:
I think we can just pass `None` for `min_new_tokens` in `SglGen` https://github.com/sgl-project/sglang/pull/1573/files#diff-192192a73b4db4b25f305dc49a35aeb3e86789d78b07c01b589243e2f2da9b18R145 for `gen_int`. What about `gen_string`? Do we also want to pass `None`?

Reply:
had a offline discussion. will set both to `None`